### PR TITLE
docs: Start from snapshot for drone push projection

### PR DESCRIPTION
**Pull request by johanandren**
Depends on #1047 and event writer fix P is working on over in Akka
Depends on https://github.com/akka/akka-projection/pull/1048

### CODE REVIEW

File: samples/grpc/local-drone-control-scala/src/main/scala/local/drones/Drone.scala [hunk 1]
```diff
@@ -96,6 +96,11 @@ object Drone {
         .receiveSignal { case (_, PostStop) =>
           telemetry.droneEntityPassivated()
         }
+        // #startFromSnapshot
+        .snapshotWhen { (_, event, _) =>
+          event.isInstanceOf[CoarseGrainedLocationChanged]
+        }
+    // #startFromSnapshot
```
Comment (johanandren) [inline samples/grpc/local-drone-control-scala/src/main/scala/local/drones/Drone.scala:103]: Tag for every coarse grained so we always have such a starting point

**Review by patriknw — COMMENTED** [review]
looking good, don't care about history and just use the latest makes sense for this

File: samples/grpc/local-drone-control-scala/src/main/scala/local/drones/DroneEvents.scala [hunk 2]
```diff
@@ -101,7 +101,15 @@ object DroneEvents {
...
-          envelope.event.isInstanceOf[Drone.CoarseGrainedLocationChanged]),
+          envelope.event.isInstanceOf[Drone.CoarseGrainedLocationChanged])
+        // #startFromSnapshot
+        // start from latest drone snapshot and don't replay history
+        .withStartingFromSnapshots[
+          Drone.State,
+          Drone.CoarseGrainedLocationChanged](state =>
+          Drone.CoarseGrainedLocationChanged(
+            state.coarseGrainedCoordinates.get)),
+      // #startFromSnapshot
```
Comment (johanandren) [inline samples/grpc/local-drone-control-scala/src/main/scala/local/drones/DroneEvents.scala:112]: This doesn't work, needs to be below where the source provider is created

**Comment by patriknw** [comment]
I added eventsBySlicesStartingFromSnapshots in the other place.
I have verified that the sample works as expected with https://github.com/akka/akka-projection/pull/1048  and https://github.com/akka/akka/pull/32191

File: samples/grpc/local-drone-control-scala/src/main/scala/local/drones/DroneEvents.scala [hunk 3]
```diff
@@ -62,12 +62,21 @@ object DroneEvents {
       R2dbcProjection.atLeastOnceFlow[Offset, EventEnvelope[Drone.Event]](
         ProjectionId("drone-event-push", s"0-$maxSlice"),
         settings = None,
-        sourceProvider = EventSourcedProvider.eventsBySlices[Drone.Event](
-          system,
-          R2dbcReadJournal.Identifier,
-          eventProducer.eventProducerSource.entityType,
-          0,
-          maxSlice),
+        // FIXME EventProducerSettings.EventProducerSource.startingFromSnapshots not used and no public accessor to pick it up from there, confusing.
```
Comment (patriknw) [inline samples/grpc/local-drone-control-scala/src/main/scala/local/drones/DroneEvents.scala:65]: Handled this fixme in https://github.com/akka/akka-projection/pull/1049

**Comment by patriknw** [comment]
Need this in the java sample too, I can add that after lunch.

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by johanandren** [comment]
Thanks for finishing it up.

**Pull request merged**